contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -596,6 +596,16 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
         return _checkIsAllowedRelayer(delivererAddress, allowedRelayers);
     }
 
+    /**
+     * @dev Sets the value of `blockchainID` to the value determined by the warp messenger precompile.
+     */
+    function initializeBlockchainID() public {
```

Comment:
With the initialization check in `receiveCrossChainMessage` do we even need `initializeBlockchainID` to be made public? The one use case is if a chain wants to intiialize it before first receive.

Reply:
Then again, we might not to attribute the additional gas costs of `WARP_MESSENGER.getBlockchainID()` to the first relayer.

Reply:
That's an interesting point. We could even remove it all and just call `WARP_MESSENGER.getBlockchainID()` if necessary in `receiveCrossChainMessage`, effectively only having lazy initialization. I'm fine with this approach because it minimizes the code changes.

Curious what you and @cam-schultz  think

Reply:
The `getBlockChainID` gas cost is very very very small also: https://github.com/ava-labs/subnet-evm/blob/master/x/warp/contract.go#L25C2-L25C24

Reply:
Nobody's going to care about 2 gas, but somebody is certainly going to forget to get confused about `initializeBlockchainID`. I'm in favor of hiding it behind the interface.

Reply:
yup I was only worried about gas incurred by first relayer. Since it's so low, I'm good with just initialization in `receiveCrossChainMessage`